packages/cadl-python/src/emitter.ts

```diff
@@ -617,9 +624,32 @@ function constantType(value: any, valueType: string): Record<string, any> {
   return { type: "constant", value: value, valueType: { type: valueType } };
 }
 
+function emitCredential(auth: HttpAuth): Record<string, any> {
+  let credential_type: Record<string, any> = {};
+  if (auth.type === "oauth2") {
+    credential_type = {
+      type: "OAuth2",
+      policy: {
+        type: "BearerTokenCredentialPolicy",
```

Comment:
We can't get info to distinguish BearerTokenCredentialPolicy and `ARMChallengeAuthenticationPolicy`. Maybe we can get it in sidecar in the future?

Reply:
good point, we basically need sidecar / additional cadl improvements to know if something is `--azure-arm` or not. I think we can do `BearerTokenCredentialPolicy` for now, but ARM is blocked from using our CADL SDKs until we add this support.

cc @Azure/linux-sme